CPDShell/Core/algorithms/DensityBasedCPD/abstracts/density_based_algorithm.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Callable
+
+import numpy as np
+from scipy.optimize import minimize
+from sklearn.neighbors import KernelDensity
+
+from CPDShell.Core.algorithms.abstract_algorithm import Algorithm
+
+
+class DensityBasedAlgorithm(Algorithm):
```

Comment:
Oh, I missed it. IDE says: "Class DensityBasedAlgorithm must implement all abstract methods". I don't see `detect` and `localize` here at all.